common/client-libs/validator-client/src/nymd/mod.rs

```diff
@@ -44,7 +44,7 @@ pub mod wallet;
 
 pub use traits::{VestingQueryClient, VestingSigningClient};
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
Why would we ever need/want to clone our client? I think we only ever want to have a single connection to the blockchain from our application, no?

Reply:
I wanted to be able to hand out clients as needed from the application state, [like so](https://github.com/nymtech/nym/blob/feature/vesting-to-wallet/nym-wallet/src-tauri/src/state.rs#L12). That used to pass references around, but in that case we needed two lines of code to pass it out, one to get the state, and one to hand out the client. This way we can do it all in one line. I like your conceptual point tough, I'll think this one through some more